sources/core/Stride.Core.CompilerServices/Common/SymbolExtensions.cs

```diff
@@ -0,0 +1,50 @@
+namespace Stride.Core.CompilerServices.Common;
+internal static class SymbolExtensions
+{
+    public static bool IsVisibleToSerializer(this ISymbol symbol, INamedTypeSymbol dataMemberAttribute)
+    {
+        if (symbol.HasAttribute(dataMemberAttribute))
+            return IsVisibleToSerializer(symbol, true);
+        return IsVisibleToSerializer(symbol, false);
+    }
+
+    public static bool IsVisibleToSerializer(this ISymbol symbol, bool hasDataMemberAttribute)
+    {
+        var accessibility = symbol.DeclaredAccessibility;
+
+        if (hasDataMemberAttribute)
+            return accessibility == Accessibility.Public || accessibility == Accessibility.Internal || accessibility == Accessibility.ProtectedOrInternal;
+        return accessibility == Accessibility.Public || accessibility == Accessibility.Internal;
```

Comment:
Are you sure about this?
Isn't only `public` visible without the attribute?

Reply:
thanks